### ISSUE TITLE
Call For Contributors - Discussion

### ISSUE THREAD

**Issue by ashleysommer** [huge-success/sanic#1907]
# Sanic Web Framework - Call for Contributors

While some industry analysts have observed an [increase in activity on open source projects](https://analyticsindiamag.com/why-open-source-might-see-higher-adoption-during-covid-19-crisis/) during the current pandemic, unfortunately that is not the case for all open source projects.

Since its [initial public release in 2016](https://news.ycombinator.com/item?id=12713469), the [Sanic Web Framework](https://github.com/huge-success/sanic) has had several waves of interest and short bursts of contribution activity, with just a handful of dedicated long-term contributors forming a core-contributors group.

In January 2020 the Sanic organization [announced a new Organization Structure](https://sanicframework.org/2020-release-managers-and-a-new-oraganizational-structure) adopting a [PEP-8016](https://www.python.org/dev/peps/pep-8016/) model including electing a new Release-Managers team, a new Steering-Council, and writing our [S.C.O.P.E](https://sanicframework.org/pages/scope.html).

Unfortunately since March 2020, both of our release managers, two of the steering council members, and three core contributors have left the project due to pressures from the pandemic and other commitments. While we're disappointed to see them leave, we wish them and their families well and all the best in their future endeavors.

The Sanic Web Framework is [more popular now than ever](https://pypistats.org/packages/sanic), with thousands of downloads from PyPI daily, Sanic is used in [almost 3500](https://github.com/huge-success/sanic/network/dependents) other open source projects on GitHub. 

The steering council has made a decision to put out a call for contributors. Sanic is a project of, and a product of, the **_community_**; everyone has a voice and the ability to help make decisions. The goal of the Sanic project is to be _community built_. The strength and direction of the project is _decision by consensus_ as laid out in the [S.C.O.P.E](https://sanicframework.org/pages/scope.html).

Doesn't matter if you have a masters in computer science, are a senior software engineer, an intern, or are a high school student. The great thing about open source is _anybody_ can contribute.

### Where can I chat to the team about this?
We have a pretty active [Community Forum](https://community.sanicframework.org/) as well as the [Gitter chatroom](https://gitter.im/sanic-python/Lobby). If you prefer to use Github, you can reply to [this issue](https://github.com/huge-success/sanic/issues/1907). 

### Why should I contribute? I don't know how to contribute! What can I even offer?
Working on a _bleeding edge_ web framework is excellent material for your resume and you learn a lot. Many of us got awesome jobs because of our open source commitments.
See the Opensource Guide's article on [Why Contribute to Open Source?](https://opensource.guide/how-to-contribute/)

#### TL;DR?
A large portion of the contributors to the Sanic project are no longer able to help, we need _your_ help.

**Comment by ashleysommer** [comment]
We need to put together a list of low hanging fruit that new contributors could help with.
  * Updating Documentation - Both in-code and online
  * Writing examples - contributing to the design and content of our new Examples Documentation online manual.
  * Issue triage
  * Making PRs to fix "good-first-issue" issues.

**Comment by ahopkins** [comment]
Great job @ashleysommer. I'll take another swing through issues to try and identify some go entry points.

**Comment by Tronic** [comment]
> This is a draft of the Call of Contributors I'm putting together. The plan is to post it on Reddit and other Social Media platforms.

Great! Overall very good announcement, my suggested diff below.

> * If you've used Sanic, liked the user experience and application performance, and would like to contribute, please think about it.
> * If you've used Sanic and didn't like the experience, please contribute to make it better for everyone.
> * If you've never even heard of Sanic, now's your chance to get to know the framework.
> * On a smaller project like this with few active contributors, you could even help to steer the direction of the project.

I'd remove this part because because it tries to target everybody and ends up saying very little.

> ### Why should I contribute? I don't know how to contribute! What can I even offer?

* Working on a *bleeding edge* web framework is excellent material for your resume and you learn a lot. Many of us got awesome jobs because of our open source commitments.

**Comment by ashleysommer** [comment]
Hi @Tronic 
> I'd remove this part because because it tries to target everybody and ends up saying very little.
Agreed, I've removed it. Any other suggestions of something to put in that part would be welcome.

**Comment by ahopkins** [comment]
I would suggest something like this:

```
Sanic is a project of the community; everyone has a voice and the ability to help make decisions.
```

Maybe not those words exactly, but that sentiment.

Once this is published I can pin it on the forums.

**Comment by ashleysommer** [comment]
>Sanic is a project of the community; everyone has a voice and the ability to help make decisions.

I've added that, and I've changed some Australian English to American English.. Endeavour to Endeavor, Organisation to Organization. I think that makes it more international. Probably not a big deal.

The https://pypistats.org/ link is still down, I hope the site isn't dead.

**Comment by ahopkins** [comment]
One of the things that I think we need to highlight is that the goal of the project is to be "community" built. The strength and direction of the project is "decision by consensus" as laid out in the SCOPE.

**Comment by yunstanford** [comment]
@ahopkins @ashleysommer Thanks for stepping up and taking the role! Yeah, pretty busy recently with limited bandwidth to fully support, but I'll still try to contribute to the project in any aspect.

**Comment by ahopkins** [comment]
@yunstanford I am super glad to hear it, and see you posting here. You've been a large part of this project and look forward to your input when you are able to provide it. I sent you some emails privately, if you get a chance to respond.

**Comment by ahopkins** [comment]
@ashleysommer when are you publishing this?

**Comment by ashleysommer** [comment]
Probably sometime Monday, unless there are any objections or suggestions of a better time to do it.

**Comment by ashleysommer** [comment]
Reddit post: https://www.reddit.com/r/Python/comments/i7pgn4/call_for_contributors_sanic_web_framework/

I created a new reddit account to post this, but /r/python doesn't accept posts from accounts less than two weeks old, so I used my primary reddit account instead, shouldn't be too much of an issue.

**Comment by krishnanunnir** [comment]
Hey, from the Reddit post, I would love to help out!

**Comment by adylanrff** [comment]
Hey, I found your post in reddit. I would really keen to help you out!

**Comment by ahopkins** [comment]
@krishnanunnir and @adylanrff thank you so much. Can you head on over to the community forums and send me a PM there? I can help get you onboarded, etc.

FYI - [topic posted on the forums](https://community.sanicframework.org/t/call-for-contributors/655)

**Comment by ashleysommer** [comment]
Something else I'll mention for any newcomers. You don't have to go through the registration process to register your email and choose a username to post on our Community Forums, you can sign in using your Github account.

**Comment by hamzabouissi** [comment]
hey, I will be glad to contribute and help .......

**Comment by AjayDextrous** [comment]
Hey there, would like to help!

**Comment by leotada** [comment]
Hello, I would like to help, when possible.

**Comment by ahopkins** [comment]
Thanks to everyone that has expressed some interest. If you have not already reached out to me on the forums, please do so.

**Comment by kckr** [comment]
Hey, I want in

**Comment by ashleysommer** [comment]
Hi @kckr Thanks for coming along to help.
Can you please post on our community forums at https://community.sanicframework.org/t/call-for-contributors/655 or contact @ahopkins (or myself) on the forums to chat about how you can help.

**Comment by ahopkins** [comment]
I posted a message on [how to get involved](https://community.sanicframework.org/t/getting-involved-in-sanic/677/2). Please take a look and let either myself or @ashleysommer know if you have questions.

Thank you to everyone that has expressed interest.

**Comment by whiz-Tuhin** [comment]
Hi, 
I'm very new to open source and would really like to contribute to the project. 
I was previously working with Java-Spring-Hibernate stack, but I'm looking to learning more about the web via a different tech stack altogether.
It would be really helpful if you could direct me to resources on how to begin.

**Comment by ahopkins** [comment]
> I posted a message on [how to get involved](https://community.sanicframework.org/t/getting-involved-in-sanic/677/2). Please take a look and let either myself or @ashleysommer know if you have questions.

[See above](https://github.com/huge-success/sanic/issues/1907#issuecomment-674448482)

**Comment by GoluffyxD** [comment]
Hey, I would love to help.
